erts/emulator/test/nif_SUITE_data/nif_mod.c

```diff
@@ -161,15 +175,31 @@ static void open_resource_type(ErlNifEnv* env, int arity, const ERL_NIF_TERM* ar
         ErlNifResourceTypeInit init;
         init.dtor = dtor;
         init.stop = NULL;
-        if (enif_is_identical(arr[6], am_null)) {
-            init.down = NULL;
-        }
-        else {
-            CHECK(enif_is_identical(arr[6], am_resource_down_D));
-            init.down = resource_down_D;
+	init.down = NULL;
+
+#  ifdef HAVE_ENIF_DYNAMIC_RESOURCE_CALL
+	init.members = 0xdead;
+	init.dyncall = (ErlNifResourceDynCall*) 0xdeadbeaf;
+
+	if (enif_is_identical(arr[6], am_resource_dyncall)) {
+            init.dyncall = resource_dyncall;
+	    init.members = 4;
```

Comment:
Should we provide a macro that does this initialization? i.e. something 

```c
#define ERL_NIF_INIT_RESOURCE_TYPE(init) \
  memset(init,0,sizeof(ErlNifResourceTypeInit)); \
  init->members = CURRENT_MAX_MEMBERS
```

Reply:
Yes maybe, but that doesn't help if you have a static initialized struct

`ErlNifResourceTypeInit my_resource_init = { my_dtor, ... };
`

Reply:
true...